### PR TITLE
ST: New test cases for Strimzi upgrade ST

### PR DESCRIPTION
**Pull request by serrss**
### Type of change
- New test cases

### Description
Added release new versions to Strimzi upgrade ST:
Upgrade from "0.11.1" to "0.11.4"
Upgrade from "0.11.4" to "0.12.0-rc1"
Upgrade from "0.11.4" to "HEAD"

### Checklist
- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by serrss** [comment]
@strimzi-ci run tests testcase=StrimziUpgradeST

**Comment by serrss** [comment]
@strimzi-ci run tests testcase=StrimziUpgradeST

**Comment by serrss** [comment]
@strimzi-ci run tests testcase=StrimziUpgradeST

**Comment by serrss** [comment]
This PR can be also cherry-picked to the release branch `release-0.12.x`

File: systemtest/src/test/java/io/strimzi/systemtest/StrimziUpgradeST.java [hunk 1]
```diff
@@ -268,6 +269,10 @@ void tearDownEnvironmentAfterEach() {
         deleteNamespaces();
     }
 
+    @Override
+    void tearDownEnvironmentAfterAll() {
+    }
```
Comment (tombentley) [inline systemtest/src/test/java/io/strimzi/systemtest/StrimziUpgradeST.java]: You might want to consider a comment about why this is overridden to do nothing.

Comment (serrss) [reply systemtest/src/test/java/io/strimzi/systemtest/StrimziUpgradeST.java]: Thanks, added comment

**Pull request merged**